### ISSUE TITLE
DataTable Filter unreliable with big tables and fast typing

### ISSUE THREAD

**Issue by Maykin-99** [primefaces/primereact#1666]
### There is no guarantee in receiving an immediate response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeReact PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**
Please fork the codesandbox below and create a case demonstrating your bug report. Issues without a codesandbox have much less possibility to be reviewed.

https://codesandbox.io/s/primereact-test-forked-9ksj1?file=/src/index.js

**Current behavior**
When trying to filter a `DataTable` that has more than a few rows and if you're not a slow typer then apparently only the first and last character is registered.

**Expected behavior**
Filtering should work reliably no matter how fast you type.

**Minimal reproduction of the problem with instructions**
Check the CodeSandbox, focus the filter field and type "123". Only "13" get's registered.

This issue is new in PrimeReact 5.x.
Previously in PrimeReact 4.x the filters in the `DataTable` were apparently uncontrolled.
Now the `value` of the `InputText` comes from `this.state.filterValue` which is set in two places:
Once in [`onFilterChange`](https://github.com/primefaces/primereact/blob/c70133ed8de6a3af42316ea86c40fa2dc600a896/src/components/datatable/HeaderCell.js#L59) and once in [`getDerivedStateFromProps`](https://github.com/primefaces/primereact/blob/c70133ed8de6a3af42316ea86c40fa2dc600a896/src/components/datatable/HeaderCell.js#L108).
The issue is apparently the latter which overrides whatever `onFilterChange` sets if `onFilterChange` is called too often.

**Please tell us about your environment:**
Linux

* **React version:**
17.0.1

* **PrimeReact version:**
5.0.1

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
Firefox 82.02
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by carltondraught** [comment]
Yep for what it's worth i'm seeing this too, makes the DataTable much less nice to use. 

Following the comment in the original issue text,  (thanks @Maykin-99 ), after replacing the content of the above-mentioned `getDerivedStateFromProps` function with `return {}` i observe the correct behavior: keystrokes are merely delayed by the filter / sort compute, not dropped.

After some testing I can't see any regression in our specific app so I suppose we are not using its intended purpose.. if i encounter i bug later ill post it here.

**Comment by cagataycivici** [comment]
We'll review, thank you.

**Comment by mcandu** [comment]
I’m unable to replicate with the latest PrimeReact. If the issue still persists with the new version please create a new issue with our code sandbox template. https://codesandbox.io/s/qjx332qq4
